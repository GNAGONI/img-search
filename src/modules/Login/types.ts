import { StateStatus } from '../../types';

type UserState = {
  status: StateStatus;
  username: string | null;
  isAuthenticated: boolean;
}

type LoginPayload = { 
  username: string;
}

export type {
  UserState,
  LoginPayload,
};
