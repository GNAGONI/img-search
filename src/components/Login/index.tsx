import { useState, useEffect, SyntheticEvent } from 'react';
import { useNavigate } from 'react-router-dom';
import { useSelector, useDispatch } from 'react-redux';
import { getUserIsAuthenticated } from './state/selectors';
import { loginRequest } from './state/slice';

const Login = () => {
  const navigate = useNavigate();
  const dispatch = useDispatch()
  const [username, setUsername] = useState('');
  const isAuthenticated = useSelector(getUserIsAuthenticated);

  const handleSubmit = (e: SyntheticEvent) => {
    e.preventDefault();
    dispatch(loginRequest({ username }))
  };

  useEffect(() => {
    if (isAuthenticated) {
      navigate('/home');
    }
  }, [isAuthenticated, navigate]);

  return (
    <div>
      <form onSubmit={handleSubmit}>
        <input
          type="text"
          value={username || ''}
          onChange={(e: SyntheticEvent) => setUsername((e.target as HTMLTextAreaElement).value)}
        />
        <button type="submit" color="primary">
          Log in
        </button>
      </form>
    </div>
  );
};

export default Login;
