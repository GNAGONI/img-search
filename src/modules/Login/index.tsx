import { useState, useEffect, SyntheticEvent } from 'react';
import { useNavigate } from 'react-router-dom';
import { useSelector } from 'react-redux';
import { getUserIsAuthenticated } from './state/selectors';
import { loginRequest } from './state/slice';
import Button from '../../components/Button';
import TextField from '../../components/TextField';
import Form from '../../components/Form';
import Wrapper from '../../components/Wrapper';
import useAppDispatch from '../../hooks/useAppDispatch';

const Login = () => {
  const navigate = useNavigate();
  const dispatch = useAppDispatch();
  const [username, setUsername] = useState('');
  const isAuthenticated = useSelector(getUserIsAuthenticated);

  const handleSubmit = (e: SyntheticEvent) => {
    e.preventDefault();
    dispatch(loginRequest({ username }));
  };

  useEffect(() => {
    if (isAuthenticated) {
      navigate('/home');
    }
  }, [isAuthenticated, navigate]);

  return (
    <Wrapper orientation='column'>
      <Form onSubmit={handleSubmit}>
        <TextField
          label="Username"
          value={username || ''}
          onChange={(e: SyntheticEvent) => setUsername((e.target as HTMLTextAreaElement).value)}
        />
        <Button title="Log in" />
      </Form>
    </Wrapper>
  );
};

export default Login;
