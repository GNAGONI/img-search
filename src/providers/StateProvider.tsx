import React, { ReactElement } from 'react';
import { Provider } from 'react-redux';
import createSagaMiddleware from 'redux-saga';
import { configureStore } from '@reduxjs/toolkit';
import logger from 'redux-logger';
import { all } from 'redux-saga/effects';
import userSagas from '../components/Login/state/sagas';
import userReducer from '../components/Login/state/slice';
import homeSagas from '../components/Home/state/sagas';
import homeReducer from '../components/Home/state/slice';


function* rootSaga() {
  yield all([
    ...userSagas,
    ...homeSagas,
  ]);
}

const sagaMiddleware = createSagaMiddleware();

type Props = {
  children: ReactElement,
}

const setupStore = () => {
  const store = configureStore({
    reducer: {
      user: userReducer,
      home: homeReducer,
    },
    middleware: (getDefaultMiddleware) => getDefaultMiddleware({ thunk: false })
      .concat(sagaMiddleware) // TODO: check for prepend
      .concat(logger),
  });
  return store;
}

const store = setupStore();

const StateProvider = ({ children }: Props) => {
  sagaMiddleware.run(rootSaga);

  return (
    <Provider store={store}>
      {children}
    </Provider>
  );
};

export default StateProvider;
