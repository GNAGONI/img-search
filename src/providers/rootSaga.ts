import { all } from 'redux-saga/effects';
import userSagas from '../components/Login/state/sagas';

export default function* rootSaga() {
  yield all([
    ...userSagas,
  ]);
}
