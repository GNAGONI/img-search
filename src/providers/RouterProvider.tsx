import { BrowserRouter, Routes, Route } from "react-router-dom";
import Layout from "../modules/Layout";
import Login from "../modules/Login";
import Home from "../modules/Home";
import Error from "../modules/Error";
import PrivateRoute from '../modules/PrivateRoute';

const RouterProvider = () => (
  <BrowserRouter>
    <Routes>
      <Route path="/" element={<Layout />}>
        <Route index element={<Login />} />
        <Route
          path="/home"
          element={(
            <PrivateRoute>
              <Home />
            </PrivateRoute>
          )}
        />
        <Route path="*" element={<Error />} />
      </Route>
    </Routes>
  </BrowserRouter>
);

export default RouterProvider;
