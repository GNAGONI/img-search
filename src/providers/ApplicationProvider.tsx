
import Spinner from '../modules/Spinner';
import Notification from '../modules/Notification';
import StateProvider from './StateProvider';
import RouterProvider from './RouterProvider';
import ThemeProvider from './ThemeProvider';

const AppProvider = () => (
  <StateProvider>
    <ThemeProvider>
      <Spinner />
      <Notification />
      <RouterProvider />
    </ThemeProvider>
  </StateProvider>
);

export default AppProvider;
