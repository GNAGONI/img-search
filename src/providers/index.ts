import ApplicationProvider from './ApplicationProvider';
import StateProvider from './StateProvider';

export { 
  ApplicationProvider,
  StateProvider,
};