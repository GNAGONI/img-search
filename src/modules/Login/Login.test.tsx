import renderer from 'react-test-renderer';
import { BrowserRouter, Routes, Route } from "react-router-dom";
import Login from './index';
import { StateProvider } from '../../providers';


it('renders correctly Login', () => {
  const tree = renderer
    .create(
      <StateProvider>
        <BrowserRouter>
          <Routes>
            <Route index element={<Login />} />
          </Routes>
        </BrowserRouter>
      </StateProvider>
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
