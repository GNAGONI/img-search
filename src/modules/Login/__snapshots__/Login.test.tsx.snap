// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly Login 1`] = `
<div
  className="css-d5gdhc"
  orientation="column"
>
  <form
    className="css-epn0vg"
    onSubmit={[Function]}
  >
    <div
      className="MuiFormControl-root MuiTextField-root css-t7h6ca-MuiFormControl-root-MuiTextField-root"
    >
      <label
        className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined MuiFormLabel-colorPrimary Mui-required MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined css-14s5rfu-MuiFormLabel-root-MuiInputLabel-root"
        data-shrink={false}
        htmlFor=":r0:"
        id=":r0:-label"
      >
        Username
        <span
          aria-hidden={true}
          className="MuiFormLabel-asterisk MuiInputLabel-asterisk css-wgai2y-MuiFormLabel-asterisk"
        >
           
          *
        </span>
      </label>
      <div
        className="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-formControl css-9ddj71-MuiInputBase-root-MuiOutlinedInput-root"
        onClick={[Function]}
      >
        <input
          aria-invalid={false}
          autoFocus={false}
          className="MuiInputBase-input MuiOutlinedInput-input css-1t8l2tu-MuiInputBase-input-MuiOutlinedInput-input"
          disabled={false}
          id=":r0:"
          onAnimationStart={[Function]}
          onBlur={[Function]}
          onChange={[Function]}
          onFocus={[Function]}
          required={true}
          type="text"
          value=""
        />
        <fieldset
          aria-hidden={true}
          className="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
        >
          <legend
            className="css-yjsfm1"
          >
            <span>
              Username
               
              *
            </span>
          </legend>
        </fieldset>
      </div>
    </div>
    <button
      className="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium css-q1b40h-MuiButtonBase-root-MuiButton-root"
      disabled={false}
      onBlur={[Function]}
      onContextMenu={[Function]}
      onDragLeave={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      tabIndex={0}
      type="submit"
    >
      <p>
        Log in
      </p>
    </button>
  </form>
</div>
`;
